beeline/src/java/org/apache/hive/beeline/BeeLine.java

```diff
@@ -1184,14 +1184,16 @@ private void setupHistory() throws IOException {
     }
 
     this.history = new FileHistory(new File(getOpts().getHistoryFile()));
-    // add shutdown hook to flush the history to history file
+    // add shutdown hook to flush the history to history file and to close all open connections
     ShutdownHookManager.addShutdownHook(new Runnable() {
       @Override
       public void run() {
         try {
           history.flush();
         } catch (IOException e) {
           error(e);
+        } finally {
```

Comment:
This shutdown hook is being added as part of setupHistory method, using that for close is not very clean. Maybe  create a separate function for adding shutdown hook ?